### PR TITLE
MWPW-144110: rename merch-cards to merch-card-collection

### PR DESCRIPTION
**Pull request by yesil**
`merch-cards` was already merged in Milo but not used in production yet.

With this PR, we are renaming it to `merch-card-collection`.

Resolves: [MWPW-144110](https://jira.corp.adobe.com/browse/MWPW-144110)

Also includes:  

https://github.com/adobecom/milo/pull/1999 without changes in personalization.js
https://git.corp.adobe.com/wcms/tacocat.js/pull/551

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ilyas/merch-card-collection?martech=off
- After: https://mwpw-144110--milo--yesil.hlx.page/drafts/ilyas/merch-card-collection?martech=off

*Performance*:
98 pts on hlx.live:
https://pagespeed.web.dev/analysis/https-mwpw-144110--milo--yesil-hlx-live-drafts-ilyas-merch-card-collection/myeewo60ru?form_factor=mobile

**CC URLs:**
https://mwpw-143951--cc--adobecom.hlx.live/drafts/ilyas/catalog?georouting=off&milolibs=mwpw-144110--milo--yesil

### CODE REVIEW

**Review by npeltier — APPROVED** [review]
i'll probably add another round of change on this for personnalization, but i guess we are fine.

**Comment by yesil** [comment]
@npeltier ideally you can build on top of this PR your upcoming changes.

**Comment by afmicka** [comment]
Noting that the changes do not have backward compatibility and they break for example https://main--cc--adobecom.hlx.page/products/catalog?milolibs=mwpw-144110--milo--yesil&martech=off. Verifying the PR relying on the slack discussion that backwards compatibility is not needed https://adobe-mwp.slack.com/archives/C0475EYRAKA/p1711375760641259

File: libs/blocks/merch-card-collection/merch-card-collection.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.section[class*="-merch-cards"] .full-width {
```
Comment (3ch023) [inline libs/blocks/merch-card-collection/merch-card-collection.css:1]: isn't the new class 'merch-card-collection'?

**Review by 3ch023 — APPROVED** [review]

File: libs/blocks/merch-card-collection/merch-card-collection.css (hunk 1 above)
Comment (yesil) [reply libs/blocks/merch-card-collection/merch-card-collection.css:1]: these are style tokens for Milo section, e.g:  `one merch card, two merch cards, three merch cards, four merch cards`.

Comment (3ch023) [reply libs/blocks/merch-card-collection/merch-card-collection.css:1]: got it, thx

**Pull request merged**